File: js/julia/colors.ts
export const black = (data: Array<number>) => {
  data.push(0);
  data.push(0);
  data.push(0);
  data.push(255);
};

export const blueToMagenta = (data: Array<number>, d: number) => {
  data.push(63 + 6 * d);
  data.push(63);
  data.push(255);
  data.push(255);
};

export const magentaToWhite = (data: Array<number>, d: number) => {
  data.push(255);
  data.push(63 + 6 * (d - 32));
  data.push(255);
  data.push(255);
};

export const whiteToCyan = (data: Array<number>, d: number) => {
  data.push(255 - 6 * (d - 64));
  data.push(255);
  data.push(255);
  data.push(255);
};

export const cyanToGreen = (data: Array<number>, d: number) => {
  data.push(63);
  data.push(255);
  data.push(255 - 6 * (d - 96));
  data.push(255);
};

export const greenToYellow = (data: Array<number>, d: number) => {
  data.push(63 + 6 * (d - 128));
  data.push(255);
  data.push(63);
  data.push(255);
};

export const yellowToRed = (data: Array<number>, d: number) => {
  data.push(255);
  data.push(255 - 6 * (d - 160));
  data.push(63);
  data.push(255);
};

export const redToBlack = (data: Array<number>, d: number) => {
  data.push(255 - 6 * (d - 192));
  data.push(63);
  data.push(63);
  data.push(255);
};

export const blackToBlue = (data: Array<number>, d: number) => {
  data.push(63);
  data.push(63);
  data.push(63);
  data.push(63 + 6 * (d - 224));
};

File: src/julia/colors.rs
pub fn black(data: &mut Vec<u8>) {
  data.push(0);
  data.push(0);
  data.push(0);
  data.push(255);
}

pub fn blue_to_magenta(data: &mut Vec<u8>, d: u32) {
  data.push((63 + 6 * d) as u8);
  data.push(63);
  data.push(255);
  data.push(255);
}

pub fn magenta_to_white(data: &mut Vec<u8>, d: u32) {
  data.push(255);
  data.push((63 + 6 * (d - 32)) as u8);
  data.push(255);
  data.push(255);
}

pub fn white_to_cyan(data: &mut Vec<u8>, d: u32) {
  data.push((255 - 6 * (d - 64)) as u8);
  data.push(255);
  data.push(255);
  data.push(255);
}

pub fn cyan_to_green(data: &mut Vec<u8>, d: u32) {
  data.push(63);
  data.push(255);
  data.push((255 - 6 * (d - 96)) as u8);
  data.push(255);
}

pub fn green_to_yellow(data: &mut Vec<u8>, d: u32) {
  data.push((63 + 6 * (d - 128)) as u8);
  data.push(255);
  data.push(63);
  data.push(255);
}

pub fn yellow_to_red(data: &mut Vec<u8>, d: u32) {
  data.push(255);
  data.push((255 - 6 * (d - 160)) as u8);
  data.push(63);
  data.push(255);
}

pub fn red_to_black(data: &mut Vec<u8>, d: u32) {
  data.push((255 - 6 * (d - 192)) as u8);
  data.push(63);
  data.push(63);
  data.push(255);
}

pub fn black_to_blue(data: &mut Vec<u8>, d: u32) {
  data.push(63);
  data.push(63);
  data.push(63);
  data.push((63 + 6 * (d - 224)) as u8);
}
